### PR TITLE
Refactor FXIOS-10645 Update Bitrise to use XCode 16.1

### PR DESCRIPTION
**Pull request by OrlaM**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :bulb: Description
Updating Bitrise to use XCode 16.1

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by clarmso** [comment]
Both Firefox and Focus fail on the `swift run` step. 😞 
```
./content_blocker_update.sh: line 17:  6485 Killed: 9               swift run
```

File: checkout.sh [hunk 1]
```diff
@@ -18,4 +18,6 @@ curl --proto '=https' --tlsv1.2 -sSf  https://raw.githubusercontent.com/mozilla/
 cd .. # Make sure we are at the root of the repo
 rm -rf shavar-prod-lists && git clone https://github.com/mozilla-services/shavar-prod-lists.git && git -C shavar-prod-lists checkout $SHAVAR_COMMIT_HASH
 
-(cd BrowserKit && swift run)
+cd BrowserKit
+swift run || true
+swift run
```
Comment (clarmso) [inline checkout.sh:23]: This workaround works on Github Actions as well. 🎉

**Comment by OrlaM** [comment]
We have a green build 🙌

**Comment by OrlaM** [comment]
@mdotb-moz or @clarmso I can merge this as soon as I get the ok from the automation side.

**Comment by OrlaM** [comment]
@MergifyIO backport release/v133

**Comment by isabelrios** [comment]
We may need the fix in this [PR](https://github.com/mozilla-mobile/firefox-ios/pull/23290) here so that the failing test passes

**Review by mdotb-moz — APPROVED** [review]
LGTM

**Pull request merged**